### DISCUSSION TITLE
Algum material de arquitetura mais avançadas para Frontend End?

### DISCUSSION THREAD

**Discussion by wilsonneto-swe** [frontendbr/forum#2041] [Discussão]
Galera, alguém ai tem dicas e materiais de arquiteturas mais avançadas para Frontend End?

Estou buscando uma arquitetura onde eu consiga isolar o React ou Vue em minhas aplicações apenas para tratar os views mesmo, onde a ação de um clique por exemplo fique em um service separado, claro que não para links, mas apenas para botões que são gatilhos de lógicas.

Alguem indica algum material?

**Comment by lddv** [comment]
Me parece um caso para aplicar [Smart vs Dumb components](https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0). Sugiro usar o [Storybook](https://storybook.js.org/) pra fazer a visualização dos componentes.

Já a arquitetura, acredito que as referências clássicas do backend podem ser usadas no front, como [princípios SOLID](https://en.wikipedia.org/wiki/SOLID). Vai depender muito do seu caso.

Infelizmente, todos os links acima estão em inglês, mas espero que ajude.
